astropy/units/quantity_helper/erfa.py

```diff
@@ -73,18 +97,250 @@ def helper_gd2gc(f, nounit, unit1, unit2, unit3):
+
+
+def helper_apio(f, unit_sp, unit_theta, unit_elong, unit_phi, unit_hm,
+                unit_xp, unit_yp, unit_refa, unit_refb):
+    from astropy.units.si import radian, m
+    return [get_converter(unit_sp, radian),
+            get_converter(unit_theta, radian),
+            get_converter(unit_elong, radian),
+            get_converter(unit_phi, radian),
+            get_converter(unit_hm, m),
+            get_converter(unit_xp, radian),
+            get_converter(unit_xp, radian),
+            get_converter(unit_xp, radian),
+            get_converter(unit_xp, radian)], astrom_unit()
+
+
+def helper_atciq(f, unit_rc, unit_dc, unit_pr, unit_pd, unit_px, unit_rv, unit_astrom):
+    from astropy.units.si import radian, arcsec, year, km, s
+    return [get_converter(unit_rc, radian),
+            get_converter(unit_dc, radian),
+            get_converter(unit_pr, radian / year),
+            get_converter(unit_pd, radian / year),
+            get_converter(unit_px, arcsec),
+            get_converter(unit_rv, km / s),
+            get_converter(unit_astrom, astrom_unit())], (radian, radian)
+
+
+def helper_atciqn(f, unit_rc, unit_dc, unit_pr, unit_pd, unit_px, unit_rv, unit_astrom,
+                  unit_b):
+    from astropy.units.si import radian, arcsec, year, km, s
+    return [get_converter(unit_rc, radian),
+            get_converter(unit_dc, radian),
+            get_converter(unit_pr, radian / year),
+            get_converter(unit_pd, radian / year),
+            get_converter(unit_px, arcsec),
+            get_converter(unit_rv, km / s),
+            get_converter(unit_astrom, astrom_unit()),
+            get_converter(unit_b, ldbody_unit())], (radian, radian)
+
+
+def helper_atciqz_aticq(f, unit_rc, unit_dc, unit_astrom):
+    from astropy.units.si import radian
+    return [get_converter(unit_rc, radian),
+            get_converter(unit_dc, radian),
+            get_converter(unit_astrom, astrom_unit())], (radian, radian)
+
+
+def helper_aticqn(f, unit_rc, unit_dc, unit_astrom, unit_b):
+    from astropy.units.si import radian
+    return [get_converter(unit_rc, radian),
+            get_converter(unit_dc, radian),
+            get_converter(unit_astrom, astrom_unit()),
+            get_converter(unit_b, ldbody_unit())], (radian, radian)
+
+
+def helper_atioq(f, unit_rc, unit_dc, unit_astrom):
+    from astropy.units.si import radian
+    return [get_converter(unit_rc, radian),
+            get_converter(unit_dc, radian),
+            get_converter(unit_astrom, astrom_unit())], (radian,)*5
+
+
+def helper_atoiq(f, unit_type, unit_ri, unit_di, unit_astrom):
+    from astropy.units.si import radian
+    if unit_type is not None:
+        raise UnitTypeError("argument 'type' should not have a unit")
+
+    return [None,
+            get_converter(unit_ri, radian),
+            get_converter(unit_di, radian),
+            get_converter(unit_astrom, astrom_unit())], (radian, radian)
+
+
 def get_erfa_helpers():
```

Comment:
The registry expects a function that actually imports the helpers, which only gets called once, on first use. This is not important for `erfa`, since that is a required dependency, but for `scipy` it allows the function that returns the dict to do the import of `scipy`.

But to be honest, here one might as well just add directly to `UFUNC_HELPERS`. Which does seem a separate issue - #11816.